### PR TITLE
Add ccr follow info api

### LINKED ISSUE #37127: [CCR] Paused follower does not display in follower stats
I created a follower index, `kibana_sample_data_logs`.

Then I paused the follower index using:
```
POST /kibana_sample_data_logs/_ccr/pause_follow
```

Then I attempted to retrieve the stats of the follower index using both:
```
GET /kibana_sample_data_logs/_ccr/stats
```
and
```
GET /_ccr/stats
```

Neither one lists `kibana_sample_data_logs` as part of `indices` array. I resumed the follower using:
```
POST /kibana_sample_data_logs/_ccr/resume_follow
{}
```
and then the stats API showed the correct information.

**For CCR UI, we need the stats API to:**

1. Show paused follower indices
2. Show status for each follower index - whether it is paused or active. Could just be a boolean `paused` flag.

This will make it so that we can give the user the correct option in the UI (whether to resume or pause follower index based on its status).

cc @elastic/es-ui

### PR DESCRIPTION
**Pull request by martijnvg**
This api returns all or requested follower indices and per follower index
the provided parameters at put follow / resume follow time and
whether index following is paused or active.

This api is useful for the UI. Figuring out which index is a follower index, 
what the active parameters are for a follower index and
whether index follow is active or paused, isn't straightforward and
requires internal ccr knowledge. Having this logic in the UI is the wrong place.

Closes #37127

### CODE REVIEW

File: docs/reference/ccr/apis/follow/get-follow-info.asciidoc [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  (object) an object encapsulating ccr replication related parameters
+
+The `parameters` contains the following fields:
+
+`indices[].parameters.max_read_request_operation_count`::
+  (integer) the configured maximum number of operations to pull per read from the remote
+  cluster
+
+`indices[].parameters.max_outstanding_read_requests`::
+  (long) the configured maximum number of outstanding reads requests from the remote
```
Comment (dnhatn) [inline docs/reference/ccr/apis/follow/get-follow-info.asciidoc]: reads requests -> read requests.

File: docs/reference/ccr/apis/follow/get-follow-info.asciidoc [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+`indices[].parameters.max_write_request_size`::
+  (<<byte-units,byte value>>) the configured maximum total bytes of operations per bulk write request
+  executed on the follower
+
+`indices[].parameters.max_outstanding_write_requests`::
+  (integer) the configured maximum number of outstanding write requests on the follower
+
+`indices[].parameters.max_write_buffer_count`::
+  (integer) th configurede maximum number of operations that can be queued for writing;
```
Comment (dnhatn) [inline docs/reference/ccr/apis/follow/get-follow-info.asciidoc]: th configurede -> the configured

**Review by dnhatn — COMMENTED** [review]
Thanks @martijnvg. This looks good. I left some minor comments.

**Comment by dnhatn** [comment]
@lcawl Would you please have a look at the doc? Thank you!

**Comment by martijnvg** [comment]
Thanks @dnhatn! I've updated the PR.

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportFollowInfoAction.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    protected FollowInfoAction.Response read(StreamInput in) throws IOException {
+        return new FollowInfoAction.Response(in);
+    }
+
+    @Override
+    protected void masterOperation(FollowInfoAction.Request request,
+                                   ClusterState state,
+                                   ActionListener<FollowInfoAction.Response> listener) throws Exception {
+
+        List<String> concreteFollowerIndices = Arrays.asList(indexNameExpressionResolver.concreteIndexNames(state,
```
[the remark this answers is not in the document]
Comment (dnhatn) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportFollowInfoAction.java]: okay.

File: docs/reference/ccr/apis/follow/get-follow-info.asciidoc [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+`indices[].status`::
+  (string) Whether index following is `active` or `paused`
+
+`indices[].parameters`::
+  (object) An object that encapsulates {ccr} parameters
+
+The `parameters` contains the following fields:
+
+`indices[].parameters.max_read_request_operation_count`::
+  (integer) The configured maximum number of operations to pull per read from
```
Comment (lcawl) [inline docs/reference/ccr/apis/follow/get-follow-info.asciidoc]: I'm not sure what the purpose of "configured" is here and in other descriptions.  Does this mean it only contains that information if you've explicitly set it (i.e. it doesn't appear if you're using default values)?

Comment (martijnvg) [reply docs/reference/ccr/apis/follow/get-follow-info.asciidoc]: No, if a parameter has not been configured then the default value is returned.
So maybe 'configured' should be removed.

**Comment by martijnvg** [comment]
@elasticmachine run gradle build tests 1

**Review by dnhatn — APPROVED** [review]
LGTM.

**Pull request merged**